src/python/visclaw/ianimate.py

```diff
@@ -17,23 +17,34 @@ def ianimate(frame_list,plotdata=None,ivar=0,varname=None,**kargs):
     from clawpack.pyclaw import Controller
     import numpy as np
 
+    figsize = kargs.get('figsize', (10,6))
+    ylim = kargs.get('ylim', None)
+    xlim = kargs.get('xlim', None)
+
     if isinstance(frame_list,Controller):
         frame_list = frame_list.frames
 
     frame = frame_list[0]
     ndim = len(frame.q.shape)-1
 
     if ndim == 1:
-        fig = plt.figure(figsize=(10,6))
+        fig = plt.figure(figsize=figsize)
         ax = plt.axes()
         im, = ax.plot([], [],lw=2)
-        
+
         xc = frame.state.grid.p_centers[0]
-        ax.set_xlim(xc[0],xc[-1])
-        ymax = max([np.max(f.q[ivar,:]) for f in frame_list])
-        ymin = min([np.min(f.q[ivar,:]) for f in frame_list])
-        ydiff = ymax-ymin
-        ax.set_ylim(ymin-ydiff/10.,ymax+ydiff/10.)
+        if xlim is None:
+            ax.set_xlim(xc[0],xc[-1])
+        else:
+            ax.set_xlim(xlim)
+
+        if ylim is None:
+            ymax = max([np.max(f.q[ivar,:]) for f in frame_list])
+            ymin = min([np.min(f.q[ivar,:]) for f in frame_list])
+            ydiff = ymax-ymin
+            ax.set_ylim(ymin-ydiff/10.,ymax+ydiff/10.)
```

Comment:
I'm not sure which part you're referring to.  I wrote this code and it does what I usually want: tight axes in x, but a little extra space at the top and bottom in y (in case there is a constant state just at the top or bottom boundary).  Also, we know the x bounds immediately, but in order to avoid clipping data or auto-rescaling axes we have to look at all the frames for y.